### PR TITLE
feat: add support to register eth2 gateway via CLI

### PR DESCRIPTION
**Pull request by ahkohd**
# Summary of changes

**_Summarize_** the description of your changes.

If it fixes a bug or creates a feature, link the issue.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

Release Notes:

Chore: No changes needed.

> "No bugs to be found,
> Our code is safe and sound.
> Let's celebrate this day,
> With a smile and hooray!"

### CODE REVIEW

File: client/packages/cli/src/commands/register/vendors/substrate.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    authoritySetId: circuit.createType("SetId", authoritySetId),
+  }
+}
+
+export const fetchLatestAuthoritySetUpdateBlock = async (
+  subscanIndexerApiEndpoint: string
+) => {
+  try {
+    const response = await fetch(
+      subscanIndexerApiEndpoint + "/api/scan/events",
```
Comment (3h4x) [inline client/packages/cli/src/commands/register/vendors/substrate.ts:108]: maybe a comment to docs?
https://support.subscan.io/#events

what does row 20 mean here?

File: runtime/standalone/Cargo.toml [hunk 2]
```diff
@@ -35,6 +35,7 @@ pallet-portal-rpc-runtime-api             = { path = "../../pallets/portal/rpc/r
 pallet-xdns                               = { path = "../../pallets/xdns", default-features = false }
 pallet-xdns-rpc-runtime-api               = { path = "../../pallets/xdns/rpc/runtime-api", default-features = false }
 pallet-eth2-finality-verifier             = { git = "https://github.com/t3rn/eth2-light-client", branch = "main", default-features = false }
+#pallet-eth2-finality-verifier             = { path = "../../../eth2-light-client", default-features = false }
```
Comment (3h4x) [inline runtime/standalone/Cargo.toml:38]: if we will leave it, then merge will do runtime upgrade.
better to add it to our documentation on how to locally use `pallet-eth2-finality-verifier`

File: client/packages/cli/src/commands/register/vendors/eth.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    throw new Error(
+      "Oops! Fetch debug beacon state faulted to error status: " +
+      response.status
+    )
+  }
+
+  const responseData = (await response.json()) as [NextSyncCommitteeResponse]
+
+  const next: SyncCommittee = {
+    pubs: responseData[0].data.next_sync_committee.pubkeys,
```
[the remark this answers is not in the document]
Comment (petscheit) [reply client/packages/cli/src/commands/register/vendors/eth.ts:138]: Nope, always 1

File: client/packages/cli/.nvmrc [hunk 4]
```diff
@@ -1 +1 @@
-20.1
+16.20.0
```
[the remark this answers is not in the document]
Comment (ahkohd) [reply client/packages/cli/.nvmrc:1]: we can revert it

**Review by petscheit — APPROVED** [review]

**Pull request merged**